File: mobile/lib/models/bootstrap.dart
class BootstrapResponse {
  BootstrapResponse({
    required this.residents,
    required this.medicationOrders,
    required this.marEntries,
    required this.administrations,
    required this.cursor,
  });

  final List<Map<String, dynamic>> residents;
  final List<Map<String, dynamic>> medicationOrders;
  final List<Map<String, dynamic>> marEntries;
  final List<Map<String, dynamic>> administrations;
  final int cursor;

  static BootstrapResponse fromJson(Map<String, dynamic> json) {
    List<Map<String, dynamic>> asList(String key) {
      final raw = json[key] as List? ?? [];
      return raw.map((e) => Map<String, dynamic>.from(e as Map)).toList();
    }

    return BootstrapResponse(
      residents: asList('residents'),
      medicationOrders: asList('medication_orders'),
      marEntries: asList('mar_entries'),
      administrations: asList('administrations'),
      cursor: json['cursor'] as int,
    );
  }
}

File: server/src/handlers/bootstrap.rs
use crate::{auth::AuthUser, models::BootstrapResponse, AppState};
use axum::{extract::State, http::StatusCode, Json};

pub async fn bootstrap(
    auth: AuthUser,
    State(state): State<AppState>,
) -> Result<Json<BootstrapResponse>, (StatusCode, String)> {
    let max_seq: i64 = sqlx::query_scalar(
        "SELECT COALESCE(MAX(seq), 0) FROM audit_events WHERE org_id = $1",
    )
    .bind(auth.org_id)
    .fetch_one(&state.db)
    .await
    .map_err(internal_error)?;

    Ok(Json(BootstrapResponse {
        residents: Vec::new(),
        medication_orders: Vec::new(),
        mar_entries: Vec::new(),
        administrations: Vec::new(),
        cursor: max_seq,
    }))
}

fn internal_error(err: impl std::fmt::Display) -> (StatusCode, String) {
    (StatusCode::INTERNAL_SERVER_ERROR, err.to_string())
}
